File: src/main/java/com/wyan/service/GuhuaService.java
package com.wyan.service;

import com.github.pagehelper.PageInfo;
import com.wyan.pojo.TbGuhua;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GuhuaService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbGuhua> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbGuhua guhua);


	/**
	 * 修改
	 */
	public void update(TbGuhua guhua);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbGuhua findOne(Integer id);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbGuhua guhua, int pageNum, int pageSize);
	
}

File: src/main/webapp/js/service/GuhuaService.js
//服务层
app.service('guhuaService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../guhua/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../guhua/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../guhua/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../guhua/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../guhua/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../guhua/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../guhua/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});